relay-server/src/endpoints/attachments.rs

```diff
@@ -1,54 +1,49 @@
-use actix_web::http::Method;
-use actix_web::{App, HttpRequest, HttpResponse};
+use axum::extract::{DefaultBodyLimit, Multipart, Path};
+use axum::http::StatusCode;
+use axum::response::IntoResponse;
+use axum::routing::{post, MethodRouter};
+use relay_config::Config;
 use relay_general::protocol::EventId;
+use serde::Deserialize;
 
 use crate::endpoints::common::{self, BadStoreRequest};
-use crate::envelope::Envelope;
+use crate::envelope::{AttachmentType, Envelope};
 use crate::extractors::RequestMeta;
 use crate::service::ServiceState;
-use crate::utils::MultipartItems;
+use crate::utils;
+
+#[derive(Debug, Deserialize)]
+struct AttachmentPath {
+    event_id: EventId,
+}
 
 async fn extract_envelope(
-    request: &HttpRequest<ServiceState>,
+    config: &Config,
     meta: RequestMeta,
+    path: AttachmentPath,
+    multipart: Multipart,
 ) -> Result<Box<Envelope>, BadStoreRequest> {
-    let event_id = request
-        .match_info()
-        .get("event_id")
-        .unwrap_or_default()
-        .parse::<EventId>()
-        .map_err(|_| BadStoreRequest::InvalidEventId)?;
+    let max_size = config.max_attachment_size();
+    let items = utils::multipart_items(multipart, max_size, |_| AttachmentType::default()).await?;
 
-    let max_payload_size = request.state().config().max_attachments_size();
-    let items = MultipartItems::new(max_payload_size)
-        .handle_request(request)
-        .await
-        .map_err(BadStoreRequest::InvalidMultipart)?;
-
-    let mut envelope = Envelope::from_request(Some(event_id), meta);
+    let mut envelope = Envelope::from_request(Some(path.event_id), meta);
     for item in items {
         envelope.add_item(item);
     }
     Ok(envelope)
 }
 
-async fn store_attachment(
+async fn handle(
+    state: ServiceState,
     meta: RequestMeta,
-    request: HttpRequest<ServiceState>,
-) -> Result<HttpResponse, BadStoreRequest> {
-    let envelope = extract_envelope(&request, meta).await?;
-    common::handle_envelope(request.state(), envelope).await?;
-    Ok(HttpResponse::Created().finish())
+    Path(path): Path<AttachmentPath>,
```

Comment:
nit: Why not just
```suggestion
    Path(event_id): Path<EventId>,
```
?

Reply:
The route for this endpoint also has a `:projectId` path parameter, do we ignore that one?

Reply:
The project_id is used by `RequestMeta` to check authentication. This is also why we need a struct that matches the `event_id` path param, so that it doesn't fail deserializing when there are two parameters (it wouldn't know which one to choose).